enclave/core/oe_alloc_thread.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    // Thread-specific startup and teardown calls for the memory allocator
+    void oe_alloc_thread_startup();
+    void oe_alloc_thread_teardown();
+
+#ifdef __cplusplus
+}
+#endif
```

Comment:
```suggestion
#endif

```

Reply:
I'm going to invoke https://github.com/openenclave/openenclave/blob/master/docs/DevelopmentGuide.md, specifically this bit:

```
The project Committers prefer to fix style issues in bulk using automation, so avoid submitting PRs intended to fix only a few instances of the inconsistent style.
```

If new lines at the end of files matter, then I suggest the clang format config be set accordingly for automated checking/conversion.

Reply:
I remember in some cases format checks do catch this. Given that the checks sill miss the other cases (e.g., like this one), we need to open an issue for that.

Reply:
There's no option for this in clang-format as far as I can tell. This also does not trigger a warning in clang for snmalloc_wrapper.cpp, because since C++11, there's literally no difference as far as the compiler is concerned:

> A source file that is not empty and that does not end in a new-line character, or that ends in a new-line character immediately preceded by a backslash character before any such splicing takes place, shall be processed as if an additional new-line character were appended to the file (C++11 §2.2/1).

I would have thought there would be one for oe_alloc_thread.h, but did not find one. A linux build of OE on Ubuntu prints out 304 warnings though, so I'm not certain it would have caught my eye.

Reply:
I've added final newlines here and elsewhere. I can't find a way to get clang format to do that, so I'm leaving that for another PR.